### PR TITLE
Salesforce : add Platform events test fixes #2938

### LINKED ISSUE #2938: Salesforce : add test coverage
Add more test coverage, examples of some tests to add 

Producer:
- Passing in salesforce headers and fetching salesforce response headers
- Adding more tests using REST api
- Generating SOQL query strings using the QueryHelper

Consumer:
- REST streaming api 
- Platform events

### PR DESCRIPTION
**Pull request by zbendhiba**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by aldettinger — APPROVED** [review]

File: integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java [hunk 1]
```diff
@@ -722,21 +729,6 @@ public void setSLAExpirationDate__c(java.time.LocalDate SLAExpirationDate__c) {
         this.SLAExpirationDate__c = SLAExpirationDate__c;
     }
 
-    @XStreamConverter(MultiSelectPicklistConverter.class)
-    private Account_MyMultiselectEnum[] MyMultiselect__c;
-
-    @JsonProperty("MyMultiselect__c")
```
Comment (jamesnetherton) [inline integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: Would these field removals regress https://github.com/apache/camel-quarkus/issues/3022?

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: So maybe we have something different in the settings of Salesforce account.
I need to regenerate the files with my account to get the Account class compatible with Salesforce API responses.

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: this is the kind of error that I have if I run my tests with my account using actual code :(

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: @jamesnetherton Can you test with your account with just the commit including the test ? If you have some time

Comment (jamesnetherton) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: With your changes and the previous `Account.java` it seems to be working on my account. If I consume your changes to `Account.java` then I get the error outlined in #3022.

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: @jamesnetherton Oh thanks. So let me revert those changes on `Account.java`, and  SQL query.

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java:728]: > @jamesnetherton Oh thanks. So let me revert those changes on `Account.java`, and SQL query.

I mean the getbyQuery test

Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/generated/Account.java]: @jamesnetherton It should be fine now

**Review by jamesnetherton — APPROVED** [review]

File: integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java [hunk 2]
```diff
@@ -87,6 +87,12 @@ public void configure() throws Exception {
                     + "updateTopic=true&sObjectQuery=SELECT Id, Name FROM Account")
                             .to("seda:RawPayloadCamelTestTopic");
 
+            // it takes some time for the subscriber to subscribe, so we'll try to
+            // send repeated platform events and wait until the first one is
+            // received
+            from("timer:platform")
+                    .setBody().simple("{}")
```
Comment (ppalaga) [inline integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java:94]: What does `.setBody().simple("{}")` do?

**Review by ppalaga — APPROVED** [review]

File: integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java (hunk 2 above)
Comment (zbendhiba) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java:94]: @ppalaga is ot okay for you. Do I merge ?

Comment (ppalaga) [reply integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java:94]: Oh, sorry, yes, I approved. This was just a curious question.

**Pull request merged**